src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -507,108 +535,93 @@ bool Freeze<ConfigT>::is_chunk_available_for_fast_freeze(intptr_t* frame_sp
 
   // assert(CodeCache::find_blob(*(address*)(frame_sp - SENDER_SP_RET_ADDRESS_OFFSET)) == StubRoutines::cont_doYield_stub(), ""); -- fails on Windows
   assert(StubRoutines::cont_doYield_stub()->frame_size() == frame::metadata_words, "");
-  intptr_t* const stack_top     = frame_sp + frame::metadata_words;
-  intptr_t* const stack_bottom  = _cont.entrySP() - ContinuationHelper::frame_align_words(_cont.argsize());
-
-  int size = stack_bottom - stack_top; // in words
 
+  int ret_size = _cont_size;
```

Comment:
ret_size is the total size that you need to freeze right?  Maybe call it total_size_needed or something better.